pkg/controller/stack/flux.go

```diff
@@ -108,8 +108,8 @@ func (sess *reconcileStackSession) SetupWorkdirFromFluxSource(ctx context.Contex
 func checkFluxSourceReady(obj unstructured.Unstructured) error {
 	conditions, ok, err := unstructured.NestedSlice(obj.Object, "status", "conditions")
 	if err != nil || !ok {
-		// didn't find a []Condition, so there's nothing to indicate that it's not ready
-		return nil
+		// didn't find a []Condition, so there's nothing to indicate that it's not ready there
+		goto checkArtifact
```

Comment:
I forgot all about `goto` -- this may be the first usage I've seen in the Pulumi code base.

Reply:
Yep, not surprised; I wouldn't use `goto` very often. It's pretty rare that you _can_ use `goto`, because
 - you (quite rightly) can't skip past declarations
 - defer is usually better for cleanup

In this case, I used it because there were a couple of early returns that I needed to do an additional check instead of exiting. But doing without `goto` happens to not be too bad, it just needs a `break`, and for the second part of the condition check to be conditional (which is just as sensible as before):

```go
func checkFluxSourceReady(obj unstructured.Unstructured) error {
	conditions, ok, err := unstructured.NestedSlice(obj.Object, "status", "conditions")
	if err == nil && ok {
                for _, c0 := range conditions {
		        var c map[string]interface{}
		        if c, ok = c0.(map[string]interface{}); !ok {
			        // condition isn't the right shape, move on
			        continue
		        }
		        if t, ok, err := unstructured.NestedString(c, "type"); ok && err == nil && t == "Ready" {
			        if v, ok, err := unstructured.NestedString(c, "status"); ok && err == nil && v == "True" {
				        // found the Ready condition and it is actually ready
				        break
        			}
	        		// found the Ready condition and it's something other than ready
		        	return fmt.Errorf("source Ready condition does not have status True %#v", c)
		        }
	        }
                // found Ready=true, or didn't find the Ready condition
        }

	_, ok, err = unstructured.NestedMap(obj.Object, "status", "artifact")
        if !ok || err != nil {
		return fmt.Errorf(".status.artifact does not have an Artifact object")
        }

        return nil
}
```

I'll rewrite it to avoid pinging people's unusual-language-feature-sense :-)